src/app/core/market/order-status-notifier/order-status-notifier.service.ts

```diff
@@ -54,13 +54,17 @@ export class OrderStatusNotifierService implements OnDestroy {
   }
 
   checkOrders(newOrders: any) {
-    this.oldOrders.forEach(order => {
-      newOrders.forEach(newOrder => {
-        if (this.compareOrder(order, newOrder)) {
-          this.notifyNewStatus(newOrder);
-        }
+    if (this.oldOrders.length !== newOrders.length) {
+      this.notifyNewStatus(newOrders.reverse()[0]);
+    } else {
+      this.oldOrders.forEach(order => {
```

Comment:
This is a bit strange though, why can't you just do newOrders.forEach and then oldOrders.find() using the compare function?